Assets/Scripts/UI/IngameMenu.cs

```diff
@@ -4,64 +4,58 @@
 using TMPro;
 using UnityEngine;
 using UnityEngine.SceneManagement;
+using UnityEngine.Serialization;
 using UnityEngine.UI;
 
 public class IngameMenu : MonoBehaviour
 {
     public static string NotificationText;
 
     [SerializeField]
-    private Text _timeText;
+    private Text timeText;
 
     [SerializeField]
-    private Text _liveCountText;
+    private Text liveCountText;
 
     [SerializeField]
-    private Text _notificationTextUI;
+    private Text notificationTextUI;
 
     [SerializeField]
-    private Button _restartButton;
+    private Button restartButton;
 
     [SerializeField]
-    private GameManager _gameManager;
+    private GameManager gameManager;
 
     [SerializeField]
-    private GameObject _hero;
-
-    public void EnableRestartButton()
-    {
-        _restartButton.gameObject.SetActive(true);
-    }
+    private GameObject hero;
 
     private void Start()
     {
         NotificationText = "";
-        _notificationTextUI.text = "";
-        _restartButton.onClick.AddListener(RestartLevel);
-        _restartButton.gameObject.SetActive(false);
-
-        GameManager.TimeUp += GameManager_TimeUp;
-        Health.DeathFromDamage += Health_DeathFromDamage;
-        FallTrap.PlayerInTrap += FallTrap_PlayerInTrap;
-        BallMovement.PlayerHitBall += BallMovement_PlayerHitBall;
+        notificationTextUI.text = "";
+        restartButton.onClick.AddListener(RestartLevel);
+        restartButton.gameObject.SetActive(false);
+
+        GameManager.TimeUp += ShouldRestart;
+        Health.DeathFromDamage += ShouldRestart;
+        FallTrap.PlayerInTrap += ShouldRestart;
+        BallMovement.PlayerHitBall += ShouldRestart;
```

Comment:
Should events be unsubscribed in `OnDestroy`? Because when switching between scenes InGameMenu gets destroyed and then event is send to null reference or am I wrong?

Reply:
yes, they need to be unsubscribed